* __Step 1: Importing the Relevant Libraries__

* __Step 2: Reading Data__
    
* __Step 3: Data Inspection__
    
* __Step 4: Exploratory Data Analysis__
    
* __Step 5: Data Cleaning__
    
* __Step 6: Feature Engineering__

* __Step 7: Building Model__

###  Step 1: Importing the Relevant Libraries

In [ ]:
# Basic eda library
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

# Pre-Processing Library 
from sklearn.preprocessing import LabelEncoder
from sklearn.preprocessing import StandardScaler

# Imputer 
from sklearn.impute import KNNImputer

#
from sklearn.model_selection import train_test_split

# Metrix
from sklearn.metrics import mean_squared_error
from sklearn.metrics import r2_score

# To hide or filtering Warning for clear view
import warnings
warnings.filterwarnings('always')
warnings.filterwarnings('ignore')

#### Customising Visulization

In [ ]:
sns.set(rc = {'figure.figsize':(16,5)})
sns.set_style('whitegrid')

## Step 2: Reading Data

In [ ]:
train_df = pd.read_csv('train_Df64byy.csv')
test_df = pd.read_csv('test_YCcRUnU.csv')

In [ ]:
train_df.head()

ID City_Code  Region_Code Accomodation_Type Reco_Insurance_Type  Upper_Age  \
0   1        C3         3213            Rented          Individual         36   
1   2        C5         1117             Owned               Joint         75   
2   3        C5         3732             Owned          Individual         32   
3   4       C24         4378             Owned               Joint         52   
4   5        C8         2190            Rented          Individual         44   

   Lower_Age Is_Spouse Health Indicator Holding_Policy_Duration  \
0         36        No               X1                     14+   
1         22        No               X2                     NaN   
2         32        No              NaN                     1.0   
3         48        No               X1                     14+   
4         44        No               X2                     3.0   

   Holding_Policy_Type  Reco_Policy_Cat  Reco_Policy_Premium  Response  
0                  3.0               22              11628.0         0  
1                  NaN               22              30510.0         0  
2                  1.0               19               7450.0         1  
3                  3.0               19              17780.0         0  
4                  1.0               16              10404.0         0

In [ ]:
test_df.head()

ID City_Code  Region_Code Accomodation_Type Reco_Insurance_Type  \
0  50883        C1          156             Owned          Individual   
1  50884        C4            7             Owned               Joint   
2  50885        C1          564            Rented          Individual   
3  50886        C3         1177            Rented          Individual   
4  50887        C1          951             Owned          Individual   

   Upper_Age  Lower_Age Is_Spouse Health Indicator Holding_Policy_Duration  \
0         30         30        No              NaN                     6.0   
1         69         68       Yes               X1                     3.0   
2         28         28        No               X3                     2.0   
3         23         23        No               X3                     3.0   
4         75         75        No               X3                     NaN   

   Holding_Policy_Type  Reco_Policy_Cat  Reco_Policy_Premium  
0                  3.0                5              11934.0  
1                  3.0               18              32204.8  
2                  4.0               17               9240.0  
3                  3.0               18               9086.0  
4                  NaN                5              22534.0

## Step 3: Data Inspection

In [ ]:
train_df.shape, test_df.shape

((50882, 14), (21805, 13))

* __We have 50882 rows and 14 columns in Train set whereas Test set has 21805 rows and 13 columns.__

In [ ]:
# Sum of null values in train data
train_df.isnull().sum()

ID                             0
City_Code                      0
Region_Code                    0
Accomodation_Type              0
Reco_Insurance_Type            0
Upper_Age                      0
Lower_Age                      0
Is_Spouse                      0
Health Indicator           11691
Holding_Policy_Duration    20251
Holding_Policy_Type        20251
Reco_Policy_Cat                0
Reco_Policy_Premium            0
Response                       0
dtype: int64

In [ ]:
# Sum of null values in test data
test_df.isnull().sum()

ID                            0
City_Code                     0
Region_Code                   0
Accomodation_Type             0
Reco_Insurance_Type           0
Upper_Age                     0
Lower_Age                     0
Is_Spouse                     0
Health Indicator           5027
Holding_Policy_Duration    8603
Holding_Policy_Type        8603
Reco_Policy_Cat               0
Reco_Policy_Premium           0
dtype: int64

In [ ]:
# Categorical Data
train_categorical = train_df.select_dtypes(include=[np.object])
obj_col = [col for col in train_df.keys() if train_df[col].dtype=='O']
print('Categorical Data of train data -->', train_categorical.shape[1],'\n',  obj_col)
print('*****************************************************************************')

# Numerical Data
train_numerical = train_df.select_dtypes(include=[np.int64])
print('Numerical Data of train data -->', train_numerical.shape[1])

Categorical Data of train data --> 6 
 ['City_Code', 'Accomodation_Type', 'Reco_Insurance_Type', 'Is_Spouse', 'Health Indicator', 'Holding_Policy_Duration']
*****************************************************************************
Numerical Data of train data --> 6


In [ ]:
# Categorical Data
test_categorical = test_df.select_dtypes(include=[np.object])
obj_col_test = [col for col in test_df.keys() if test_df[col].dtype=='O']
print('Categorical Data of test data -->', test_categorical.shape[1],'\n',  obj_col_test)
print('*****************************************************************************')

# Numerical Data
test_numerical = test_df.select_dtypes(include=[np.int64])
print('Numerical Data of test data -->', test_numerical.shape[1])

Categorical Data of test data --> 6 
 ['City_Code', 'Accomodation_Type', 'Reco_Insurance_Type', 'Is_Spouse', 'Health Indicator', 'Holding_Policy_Duration']
*****************************************************************************
Numerical Data of test data --> 5


## Step 5: Data Cleaning

**Data Cleaning required in these column**
- Health Indicator        --->  __11691__
- Holding_Policy_Duration --->  __20251__
- Holding_Policy_Type     --->  __20251__

* __5.1 Lets start with Health Indicator column.__

In [ ]:
train_df['Health Indicator'].unique()

array(['X1', 'X2', nan, 'X4', 'X3', 'X6', 'X5', 'X8', 'X7', 'X9'],
      dtype=object)

In [ ]:
def imp(col):
    train_df['Health Indicator']
  
    if col=='X1':
        return 1
    elif col=='X2':
        return 2
    elif col=='X3':
        return 3
    elif col=='X4':
        return 4
    elif col=='X5':
        return 5
    elif col=='X6':
        return 6
    elif col=='X7':
        return 7
    elif col=='X8':
        return 8
    elif col=='X9':
        return 9
    else:
        return np.nan

In [ ]:
#train Data 
train_df['Health Indicator'] = train_df['Health Indicator'].apply(imp).astype(float)
train_df['Health Indicator'] = train_df['Health Indicator'].replace('nan',np.nan)
# Test Data
test_df['Health Indicator'] = test_df['Health Indicator'].apply(imp).astype(float)
test_df['Health Indicator'] = test_df['Health Indicator'].replace('nan',np.nan)
train_df['Health Indicator'].dtypes

dtype('float64')

In [ ]:
train_df['Health Indicator'].unique()

array([ 1.,  2., nan,  4.,  3.,  6.,  5.,  8.,  7.,  9.])

In [ ]:
test_df['Health Indicator'].unique()

array([nan,  1.,  3.,  2.,  5.,  4.,  7.,  6.,  9.,  8.])

* __5.2 Holding_Policy_Duration column.__

In [ ]:
train_df['Holding_Policy_Duration'].unique()

array(['14+', nan, '1.0', '3.0', '5.0', '9.0', '14.0', '7.0', '2.0',
       '11.0', '10.0', '8.0', '6.0', '4.0', '13.0', '12.0'], dtype=object)

In [ ]:
test_df['Holding_Policy_Duration'].unique()

array(['6.0', '3.0', '2.0', nan, '14+', '5.0', '1.0', '4.0', '12.0',
       '11.0', '7.0', '9.0', '13.0', '8.0', '14.0', '10.0'], dtype=object)

In [ ]:

train_df['Holding_Policy_Duration'] = train_df['Holding_Policy_Duration'].replace('14+','14')

test_df['Holding_Policy_Duration'] = test_df['Holding_Policy_Duration'].replace('14+','14')

In [ ]:
test_df['Holding_Policy_Duration'].unique()

array(['6.0', '3.0', '2.0', nan, '14', '5.0', '1.0', '4.0', '12.0',
       '11.0', '7.0', '9.0', '13.0', '8.0', '14.0', '10.0'], dtype=object)

In [ ]:
train_df['Holding_Policy_Duration'].unique()

array(['14', nan, '1.0', '3.0', '5.0', '9.0', '14.0', '7.0', '2.0',
       '11.0', '10.0', '8.0', '6.0', '4.0', '13.0', '12.0'], dtype=object)

* __5.3 Holding_Policy_Type column.__

In [ ]:
train_df['Holding_Policy_Type'].unique()

array([ 3., nan,  1.,  4.,  2.])

In [ ]:
test_df['Holding_Policy_Type'].unique()

array([ 3.,  4., nan,  1.,  2.])

In [ ]:
train_df.head()

ID City_Code  Region_Code Accomodation_Type Reco_Insurance_Type  Upper_Age  \
0   1        C3         3213            Rented          Individual         36   
1   2        C5         1117             Owned               Joint         75   
2   3        C5         3732             Owned          Individual         32   
3   4       C24         4378             Owned               Joint         52   
4   5        C8         2190            Rented          Individual         44   

   Lower_Age Is_Spouse  Health Indicator Holding_Policy_Duration  \
0         36        No               1.0                      14   
1         22        No               2.0                     NaN   
2         32        No               NaN                     1.0   
3         48        No               1.0                      14   
4         44        No               2.0                     3.0   

   Holding_Policy_Type  Reco_Policy_Cat  Reco_Policy_Premium  Response  
0                  3.0               22              11628.0         0  
1                  NaN               22              30510.0         0  
2                  1.0               19               7450.0         1  
3                  3.0               19              17780.0         0  
4                  1.0               16              10404.0         0

## __5.5 Label Encoder.__

In [ ]:
# Intiallising Encoder
encoder = LabelEncoder()

In [ ]:
train_df['City_Code'] = encoder.fit_transform(train_df['City_Code'])
test_df['City_Code'] = encoder.fit_transform(test_df['City_Code'])

In [ ]:
train_df['Accomodation_Type'] = pd.get_dummies(train_df['Accomodation_Type'],drop_first=False)
train_df['Reco_Insurance_Type'] = pd.get_dummies(train_df['Reco_Insurance_Type'],drop_first=False)
train_df['Is_Spouse'] = pd.get_dummies(train_df['Is_Spouse'],drop_first=False)

In [ ]:
test_df['Accomodation_Type'] = pd.get_dummies(test_df['Accomodation_Type'],drop_first=False)
test_df['Reco_Insurance_Type'] = pd.get_dummies(test_df['Reco_Insurance_Type'],drop_first=False)
test_df['Is_Spouse'] = pd.get_dummies(test_df['Is_Spouse'],drop_first=False)

In [ ]:
len(test_df['Health Indicator'].unique())

10

In [ ]:
len(test_df['Reco_Insurance_Type'].unique())

2

In [ ]:
test_df['Reco_Insurance_Type'].value_counts()

1    17341
0     4464
Name: Reco_Insurance_Type, dtype: int64

1    17341
0     4464

In [ ]:
len(test_df['Accomodation_Type'].unique())

2

In [ ]:
test_df['Accomodation_Type'].value_counts()

1    12090
0     9715
Name: Accomodation_Type, dtype: int64

1    12090
0     9715

## __5.6 Missing Value Treatment.__

### __KNN Imputer .__

In [ ]:
knnimputer = KNNImputer(n_neighbors=5)

In [ ]:
key_tr = train_df.keys()
key_ts = test_df.keys()

In [ ]:
train_df = knnimputer.fit_transform(train_df)
test_df = knnimputer.fit_transform(test_df)

In [ ]:
train_df=pd.DataFrame(train_df,columns=key_tr)
test_df=pd.DataFrame(test_df,columns=key_ts)

In [ ]:
train_df

ID  City_Code  Region_Code  Accomodation_Type  \
0          1.0       22.0       3213.0                0.0   
1          2.0       31.0       1117.0                1.0   
2          3.0       31.0       3732.0                1.0   
3          4.0       16.0       4378.0                1.0   
4          5.0       34.0       2190.0                0.0   
...        ...        ...          ...                ...   
50877  50878.0       30.0        845.0                0.0   
50878  50879.0       31.0       4188.0                0.0   
50879  50880.0        0.0        442.0                0.0   
50880  50881.0        0.0          4.0                1.0   
50881  50882.0       22.0       3866.0                0.0   

       Reco_Insurance_Type  Upper_Age  Lower_Age  Is_Spouse  Health Indicator  \
0                      1.0       36.0       36.0        1.0               1.0   
1                      0.0       75.0       22.0        1.0               2.0   
2                      1.0       32.0       32.0        1.0               3.2   
3                      0.0       52.0       48.0        1.0               1.0   
4                      1.0       44.0       44.0        1.0               2.0   
...                    ...        ...        ...        ...               ...   
50877                  1.0       22.0       22.0        1.0               3.0   
50878                  1.0       27.0       27.0        1.0               3.0   
50879                  1.0       63.0       63.0        1.0               2.0   
50880                  0.0       71.0       49.0        1.0               2.0   
50881                  1.0       24.0       24.0        1.0               3.0   

       Holding_Policy_Duration  Holding_Policy_Type  Reco_Policy_Cat  \
0                         14.0                  3.0             22.0   
1                          1.8                  2.0             22.0   
2                          1.0                  1.0             19.0   
3                         14.0                  3.0             19.0   
4                          3.0                  1.0             16.0   
...                        ...                  ...              ...   
50877                      4.2                  2.0             18.0   
50878                      7.0                  3.0              4.0   
50879                     14.0                  1.0             12.0   
50880                      2.0                  2.0             16.0   
50881                      2.0                  3.0             18.0   

       Reco_Policy_Premium  Response  
0                  11628.0       0.0  
1                  30510.0       0.0  
2                   7450.0       1.0  
3                  17780.0       0.0  
4                  10404.0       0.0  
...                    ...       ...  
50877               7704.0       0.0  
50878               5408.0       0.0  
50879              11374.0       0.0  
50880              28179.2       1.0  
50881              11424.0       0.0  

[50882 rows x 14 columns]

In [ ]:
len(train_df['Health Indicator'].unique())

27

In [ ]:
train_df['Health Indicator'].unique()

array([1. , 2. , 3.2, 2.8, 4. , 3. , 2.6, 6. , 5. , 1.4, 1.8, 3.4, 2.2,
       4.8, 1.6, 2.4, 3.6, 1.2, 3.8, 8. , 7. , 4.2, 4.6, 4.4, 9. , 5.2,
       5.8])

In [ ]:
np.ceil(train_df['Health Indicator'])

0        1.0
1        2.0
2        4.0
3        1.0
4        2.0
        ... 
50877    3.0
50878    3.0
50879    2.0
50880    2.0
50881    3.0
Name: Health Indicator, Length: 50882, dtype: float64

In [ ]:
len(test_df['Reco_Insurance_Type'].unique())

2

In [ ]:
len(test_df['Accomodation_Type'].unique())

2

In [ ]:
train_df.isnull().sum()

ID                         0
City_Code                  0
Region_Code                0
Accomodation_Type          0
Reco_Insurance_Type        0
Upper_Age                  0
Lower_Age                  0
Is_Spouse                  0
Health Indicator           0
Holding_Policy_Duration    0
Holding_Policy_Type        0
Reco_Policy_Cat            0
Reco_Policy_Premium        0
Response                   0
dtype: int64

In [ ]:
test_df.isnull().sum()

ID                         0
City_Code                  0
Region_Code                0
Accomodation_Type          0
Reco_Insurance_Type        0
Upper_Age                  0
Lower_Age                  0
Is_Spouse                  0
Health Indicator           0
Holding_Policy_Duration    0
Holding_Policy_Type        0
Reco_Policy_Cat            0
Reco_Policy_Premium        0
dtype: int64

In [ ]:
X=train_df.drop(['Response', 'ID', 'City_Code', 'Region_Code'],axis=1)
y=train_df['Response']

In [ ]:
X

Accomodation_Type  Reco_Insurance_Type  Upper_Age  Lower_Age  \
0                    0.0                  1.0       36.0       36.0   
1                    1.0                  0.0       75.0       22.0   
2                    1.0                  1.0       32.0       32.0   
3                    1.0                  0.0       52.0       48.0   
4                    0.0                  1.0       44.0       44.0   
...                  ...                  ...        ...        ...   
50877                0.0                  1.0       22.0       22.0   
50878                0.0                  1.0       27.0       27.0   
50879                0.0                  1.0       63.0       63.0   
50880                1.0                  0.0       71.0       49.0   
50881                0.0                  1.0       24.0       24.0   

       Is_Spouse  Health Indicator  Holding_Policy_Duration  \
0            1.0               1.0                     14.0   
1            1.0               2.0                      1.8   
2            1.0               3.2                      1.0   
3            1.0               1.0                     14.0   
4            1.0               2.0                      3.0   
...          ...               ...                      ...   
50877        1.0               3.0                      4.2   
50878        1.0               3.0                      7.0   
50879        1.0               2.0                     14.0   
50880        1.0               2.0                      2.0   
50881        1.0               3.0                      2.0   

       Holding_Policy_Type  Reco_Policy_Cat  Reco_Policy_Premium  
0                      3.0             22.0              11628.0  
1                      2.0             22.0              30510.0  
2                      1.0             19.0               7450.0  
3                      3.0             19.0              17780.0  
4                      1.0             16.0              10404.0  
...                    ...              ...                  ...  
50877                  2.0             18.0               7704.0  
50878                  3.0              4.0               5408.0  
50879                  1.0             12.0              11374.0  
50880                  2.0             16.0              28179.2  
50881                  3.0             18.0              11424.0  

[50882 rows x 10 columns]

In [ ]:
y

0        0.0
1        0.0
2        1.0
3        0.0
4        0.0
        ... 
50877    0.0
50878    0.0
50879    0.0
50880    1.0
50881    0.0
Name: Response, Length: 50882, dtype: float64

In [ ]:
Xtr = X.columns

In [ ]:
test_df

ID  City_Code  Region_Code  Accomodation_Type  \
0      50883.0        0.0        156.0                1.0   
1      50884.0       30.0          7.0                1.0   
2      50885.0        0.0        564.0                0.0   
3      50886.0       22.0       1177.0                0.0   
4      50887.0        0.0        951.0                1.0   
...        ...        ...          ...                ...   
21800  72683.0       22.0       1044.0                1.0   
21801  72684.0       30.0        266.0                1.0   
21802  72685.0        3.0       2470.0                1.0   
21803  72686.0        1.0       1676.0                0.0   
21804  72687.0       11.0       3617.0                1.0   

       Reco_Insurance_Type  Upper_Age  Lower_Age  Is_Spouse  Health Indicator  \
0                      1.0       30.0       30.0        1.0               1.8   
1                      0.0       69.0       68.0        0.0               1.0   
2                      1.0       28.0       28.0        1.0               3.0   
3                      1.0       23.0       23.0        1.0               3.0   
4                      1.0       75.0       75.0        1.0               3.0   
...                    ...        ...        ...        ...               ...   
21800                  1.0       45.0       45.0        1.0               1.0   
21801                  1.0       59.0       59.0        1.0               5.0   
21802                  1.0       74.0       74.0        1.0               3.0   
21803                  1.0       25.0       25.0        1.0               4.0   
21804                  1.0       73.0       73.0        1.0               2.0   

       Holding_Policy_Duration  Holding_Policy_Type  Reco_Policy_Cat  \
0                          6.0                  3.0              5.0   
1                          3.0                  3.0             18.0   
2                          2.0                  4.0             17.0   
3                          3.0                  3.0             18.0   
4                          6.8                  2.0              5.0   
...                        ...                  ...              ...   
21800                      4.0                  1.0             18.0   
21801                      6.0                  3.0             18.0   
21802                      8.8                  2.8              1.0   
21803                      3.0                  1.0             19.0   
21804                     14.0                  2.0              3.0   

       Reco_Policy_Premium  
0                  11934.0  
1                  32204.8  
2                   9240.0  
3                   9086.0  
4                  22534.0  
...                    ...  
21800              15884.0  
21801              21390.0  
21802              17836.0  
21803              11568.0  
21804              23040.0  

[21805 rows x 13 columns]

In [ ]:
test_df.head(1)

ID  City_Code  Region_Code  Accomodation_Type  Reco_Insurance_Type  \
0  50883.0        0.0        156.0                1.0                  1.0   

   Upper_Age  Lower_Age  Is_Spouse  Health Indicator  Holding_Policy_Duration  \
0       30.0       30.0        1.0               1.8                      6.0   

   Holding_Policy_Type  Reco_Policy_Cat  Reco_Policy_Premium  
0                  3.0              5.0              11934.0

In [ ]:
X.head(1)

Accomodation_Type  Reco_Insurance_Type  Upper_Age  Lower_Age  Is_Spouse  \
0                0.0                  1.0       36.0       36.0        1.0   

   Health Indicator  Holding_Policy_Duration  Holding_Policy_Type  \
0               1.0                     14.0                  3.0   

   Reco_Policy_Cat  Reco_Policy_Premium  
0             22.0              11628.0

## Standard Scalling

In [ ]:
scaler = StandardScaler()

In [ ]:
X = scaler.fit_transform(X)
test_df = scaler.fit_transform(test_df)

In [ ]:
X.shape

(50882, 10)

In [ ]:
X=pd.DataFrame(X,columns=Xtr)
# y=pd.DataFrame(y,columns=train_df.Response)
test_df=pd.DataFrame(test_df,columns=key_ts)

In [ ]:
y

Empty DataFrame
Columns: [0.0, 0.0, 1.0, 0.0, 0.0, 1.0, 0.0, 1.0, 0.0, 1.0, 1.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, ...]
Index: []

[0 rows x 50882 columns]

In [ ]:
## Train Test Split

In [ ]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state =101)

ValueError: Found input variables with inconsistent numbers of samples: [50882, 0]

In [ ]:
from xgboost import XGBClassifier
from sklearn.model_selection import RandomizedSearchCV, GridSearchCV
from sklearn.metrics import accuracy_score

In [ ]:
## Hyper Parameter Optimization

params={
 "learning_rate"    : [0.05, 0.10, 0.15, 0.20, 0.25, 0.30 ] ,
 "max_depth"        : [ 3, 4, 5, 6, 8, 10, 12, 15],
 "min_child_weight" : [ 1, 3, 5, 7 ],
 "gamma"            : [ 0.0, 0.1, 0.2 , 0.3, 0.4 ],
 "colsample_bytree" : [ 0.3, 0.4, 0.5 , 0.7 ]
}

In [ ]:
xgboost_model = XGBClassifier(max_depth=10)

In [ ]:
random_search=RandomizedSearchCV(xgboost_model,param_distributions=params,n_iter=5,scoring='roc_auc',n_jobs=-1,cv=5)


In [ ]:
random_search.fit(X,y)

In [ ]:
random_search.best_estimator_

In [ ]:
random_search.best_params_

In [ ]:
xgbclassifier = XGBClassifier(base_score=0.5, booster='gbtree', colsample_bylevel=1,
              colsample_bynode=1, colsample_bytree=0.7, gamma=0.3, gpu_id=-1,
              importance_type='gain', interaction_constraints='',
              learning_rate=0.1, max_delta_step=0, max_depth=15,
              min_child_weight=1, monotone_constraints='()',
              n_estimators=100, n_jobs=8, num_parallel_tree=1, random_state=0,
              reg_alpha=0, reg_lambda=1, scale_pos_weight=1, subsample=1,
              tree_method='exact', validate_parameters=1, verbosity=None)

In [ ]:
xgbclassifier.fit(X, y)

In [ ]:
pred = xgbclassifier.predict(X_test)

In [ ]:
pred

In [ ]:
score = accuracy_score(pred, y_test)
score

In [ ]:
final_predictionht = xgbclassifier.predict(test_df.drop(['ID', 'City_Code', 'Region_Code'],axis=1))

In [ ]:
final_predictionht.shape

In [ ]:
a = pd.DataFrame(final_predictionht)

In [ ]:
a[0].unique()

In [ ]:
# final_predictionht.value_counts()

In [ ]:
stop

In [ ]:
submission = pd.read_csv('sample_submission_QrCyCoT.csv')  # uploading submission file
final_predictions = xgbclassifier.predict(test_df.drop(['ID', 'City_Code', 'Region_Code'],axis=1))
submission['Response'] = final_predictions
#only positive predictions for the target variable

submission.to_csv('my_submission_xgboost.csv', index=False)

In [ ]:
my = pd.read_csv('my_submission_xgboost.csv')

In [ ]:
my['Response'].value_counts()